compiler/src/iree/compiler/GlobalOptimization/Utils.cpp

```diff
@@ -52,6 +53,39 @@ std::optional<Type> getCastElemType(Value input) {
   return castSrcElemType;
 }
 
+FailureOr<Value>
+createGenericElementwiseCastOp(OpBuilder &builder, Location loc, Value input,
+                               CastOpInterface castOp,
+                               ArrayRef<NamedAttribute> attrs) {
+  auto inputType = dyn_cast<RankedTensorType>(input.getType());
+  if (!inputType) {
+    return failure();
+  }
```

Comment:
It is very wrong if we still have unranked tensor at this stage. Can we remove the check or just add an `assert`? In this context, we can return `Value` type directly.